### PR TITLE
Add a new story for developing upgrade tests

### PR DESCRIPTION
**Pull request by psss**
Here's a draft of the user story covering upgrade tests scenario discussed today. @thozza, @psklenar, @AloisMahdal, @hhorak, please review.

### CODE REVIEW

**Comment by thozza** [comment]
@mkluson, please review as well.

**Comment by thozza** [comment]
@psss the story looks good to me as a starting point. The example seems enable us to have quite a freedom in what kind of environment variables are passed to the test, thus not forcing us to define this upfront. I don't have anything to add at this point.

File: stories/cli/run.fmf [hunk 1]
```diff
@@ -43,6 +43,38 @@ story: 'As a user I want to execute tests easily'
         - tmt run --continue
         - tmt run --abort
 
+/upgrade:
+    story:
+        As a tester I want to verify that a configured application
+        or service still correctly works after the system upgrade.
+    description: |
+        In order to enable developing tests for upgrade testing we
```
Comment (hhorak) [inline stories/cli/run.fmf]: Maybe we can be more clear that we speak about the "system tests", not unit tests that are written for each actor as well, but do not test the whole upgrade story.

**Comment by hhorak** [comment]
LGTM as well.

File: stories/cli/run.fmf [hunk 2]
```diff
@@ -43,6 +43,38 @@ story: 'As a user I want to execute tests easily'
...
+        Combining test step selection feature and defining extra
+        environment variables together with an external framework
+        to perform the actual system upgrade should give us a nice
+        way how to achieve the desired functionality. In addition,
+        we will also need to implement a way how to aggregate test
+        results from both runs (before and after the upgrade).
+    example:
+        - tmt run discover provision --image=rhel7 --output=box.json
+        - tmt run execute --environment='UPGRADE=before'
+        - upgrade-the-system box.json
```
Comment (mkluson) [inline stories/cli/run.fmf]: We have implementation of this step by STI test. Can we run it also by `tmt`?

File: stories/cli/run.fmf (hunk 1 above)
Comment (psss) [reply stories/cli/run.fmf]: I see. This should now be clarified in 7cc2580.

File: stories/cli/run.fmf (hunk 2 above)
Comment (psss) [reply stories/cli/run.fmf]: We can consider integrating it in the future but for start I'd suggest we keep it simple. Individual step implementation is already quite complex. I believe we need to stabilize these features first rather that bring too much complexity at once.

File: stories/cli/run.fmf (hunk 1 above)
Comment (hhorak) [reply stories/cli/run.fmf]: Cool, thanks.

File: stories/cli/run.fmf [hunk 3]
```diff
@@ -43,6 +43,41 @@ story: 'As a user I want to execute tests easily'
...
+
+        Combining test step selection feature and defining extra
+        environment variables together with an external framework
+        to perform the actual system upgrade should give us a nice
+        way how to achieve the desired functionality. In addition,
+        we will also need to implement a way how to aggregate test
+        results from both runs (before and after the upgrade).
+    example:
+        - tmt run discover provision --image=rhel7 --output=box.json
+        - tmt run execute --environment='UPGRADE=before'
```
Comment (thrix) [inline stories/cli/run.fmf]: isn't this a new run? we should specify `run-id` ?

Comment (psss) [reply stories/cli/run.fmf]: Right you are. We are continuing the same run. Fixed in 85dd799.

File: stories/cli/run.fmf (hunk 2 above)
Comment (mkluson) [reply stories/cli/run.fmf]: Ok, I just want to let you know we already have implementation of this step

Comment (psss) [reply stories/cli/run.fmf]: Good, thanks.

**Comment by psss** [comment]
Thanks for the review guys, merging.

**Pull request merged**